src/connector/src/base.rs

```diff
@@ -257,7 +258,23 @@ impl SplitMetaData for ConnectorState {
 pub enum ConnectorStateV2 {
```

Comment:
Looks like ConnectorStateV2 is currently an `Option<Vec<SplitImpl>>`. You can make it a type alias instead of declaring it as a type.

```
type ConnectorStateV2 = Option<Vec<SplitImpl>>;
```

And pls do not use V2 anymore.